packages/api/src/Base.ts

```diff
@@ -589,37 +592,50 @@ export default abstract class ApiBase<CodecResult, SubscriptionResult> implement
 
     decorated.creator = creator;
 
-    decorated.at = (hash: Hash | Uint8Array | string, arg?: CodecArg): C =>
+    decorated.at = (hash: Hash | Uint8Array | string, arg1?: CodecArg, arg2?: CodecArg): C =>
       onCall(
-        (arg: CodecArg) => this._rpcRx.state.getStorage([creator, arg], hash),
-        [arg]
+        (...args: CodecArg[]) => this._rpcRx.state.getStorage(
+          creator.meta.type.isDoubleMap
+            ? [creator, args]
+            : [creator, ...args],
+          hash),
+        [arg1, arg2]
       ) as C;
 
     // FIXME The unknown cast is needed since the onCall result, `C | S` cannot
     // be converted from C to the actual result required
-    decorated.hash = (arg?: CodecArg): HashResult<C, S> =>
+    decorated.hash = (arg1?: CodecArg, arg2?: CodecArg): HashResult<C, S> =>
       onCall(
-        (arg: CodecArg) => this._rpcRx.state.getStorageHash([creator, arg]),
-        [arg]
+        (...args: CodecArg[]) => this._rpcRx.state.getStorageHash(
+          creator.meta.type.isDoubleMap
+            ? [creator, args]
+            : [creator, ...args]),
+        [arg1, arg2]
       ) as unknown as HashResult<C, S>;
 
-    decorated.key = (arg?: CodecArg): string =>
-      u8aToHex(compactStripLength(creator(arg))[1]);
+    decorated.key = (arg1?: CodecArg, arg2?: CodecArg): string =>
+      creator.meta.type.isDoubleMap
+      ? u8aToHex(creator([arg1, arg2]))
```

Comment:
This really is weird the way it is done here - one type has the compact removed, the other not. The reason the underlying storage has the length is that so it can go into a StorageKey, i.e. https://github.com/polkadot-js/api/blob/master/packages/types/src/primitive/StorageKey.ts and be treated correctly.

Reply:
Okay. let me check it

Reply:
I updated the implementation of doublemap, all types use compact now.